src/test/java/unit/Workspace.java

```diff
@@ -0,0 +1,261 @@
+
+    // `terra workspace list --format=json`
+    List<UFWorkspace> listWorkspaces =
+        TestCommand.runCommandExpectSuccess(
+            new TypeReference<>() {}, "workspace", "list", "--format=json");
+
+    // check the deleted workspace is not included in the list
+    List<UFWorkspace> matchingWorkspaces =
+        listWorkspaces.stream()
+            .filter(workspace -> workspace.id.equals(createWorkspace.id))
+            .collect(Collectors.toList());
+    assertEquals(0, matchingWorkspaces.size(), "deleted workspace is not included in list");
+  }
+
+  @Test
+  @DisplayName("status, workspace list reflect workspace update")
+  void statusListReflectUpdate() throws IOException {
+    // select a test user and login
+    TestUsers testUser = TestUsers.chooseTestUserWithSpendAccess();
+    testUser.login();
+
+    // `terra workspace create --format=json --name=$name --description=$description`
+    String name = "statusListReflectUpdate";
+    String description = "status list reflect update";
+    UFWorkspace createWorkspace =
+        TestCommand.runCommandExpectSuccess(
+            UFWorkspace.class,
+            "workspace",
+            "create",
+            "--format=json",
+            "--name=" + name,
+            "--description=" + description);
+
+    // check the created workspace name and description are set
+    assertNotNull(createWorkspace.name, "create workspace name is defined");
+    assertNotNull(createWorkspace.description, "create workspace description is defined");
+
+    // `terra workspace create --format=json --name=$newName --description=$newDescription`
+    String newName = "NEW_statusListReflectUpdate";
+    String newDescription = "NEW status list reflect update";
+    TestCommand.runCommandExpectSuccess(
+        "workspace",
+        "update",
+        "--format=json",
+        "--name=" + newName,
+        "--description=" + newDescription);
+
+    // `terra status --format=json`
+    UFStatus status =
+        TestCommand.runCommandExpectSuccess(UFStatus.class, "status", "--format=json");
+
+    // check the current status reflects the update
+    assertEquals(newName, status.workspace.name, "status matches updated workspace name");
+    assertEquals(
+        newDescription,
+        status.workspace.description,
+        "status matches updated workspace description");
+
+    // `terra workspace list --format=json`
+    List<UFWorkspace> listWorkspaces =
+        TestCommand.runCommandExpectSuccess(
+            new TypeReference<>() {}, "workspace", "list", "--format=json");
+
+    // check the workspace list reflects the update
+    List<UFWorkspace> matchingWorkspaces =
+        listWorkspaces.stream()
+            .filter(workspace -> workspace.id.equals(createWorkspace.id))
+            .collect(Collectors.toList());
+    assertEquals(
+        1, matchingWorkspaces.size(), "updated workspace is included exactly once in list");
```

Comment:
optional: consider a helper function for `private static UFWorkspace checkOneListedWorkspaceMatching(Predicate<UFWorkspace> predicate)` since this seems to be a common pattern in these tests.

Reply:
Added a helper method to list and filter the workspaces.